### ISSUE TITLE
Add db classes diagram

### ISSUE THREAD

**Issue by miyconst** [Starcounter/Starcounter#3063]
Today @atifsheikh asked me about db classes diagram.

I think it would be nice to add at least some basic diagram builder into Administrator.
If the diagram builder is too much of an effort, then class navigation tool at least.

**Comment by atifsheikh** [comment]
True, this sort of tool is a must for Starcounter DB. We are facing this issue, when we hire a new developer, we have to give them instructions for our DB classes, and it takes too much time for both developers (new and old).

**Comment by miyconst** [comment]
While we are working on own tool, you can use build in Visual Studio class diagram: https://msdn.microsoft.com/en-us/library/33864ckt.aspx.

**Comment by per-samuelsson** [comment]
I've had ideas about this for really long, but no one has requested it. It might a reason we get this one going at some point soon then, now that we know there is a need.

**Comment by per-samuelsson** [comment]
Just expanding on the above: most should already be available in SQL just by querying the metadata classes. But unfortunatelly they are somewhat unpolished and doesn't really serve good as a public API at this point.

And also, here's a **super extra special hack** if you want to try to get a schema I'm producing in the weaver. It's very unsupported of course, and will not survive once we move further, possibly not even within 2.x, so if you decide to try it, take it for what it is (a hack).

To see the schema of the current application, run

```
star app.cs
```

on this:

**app.cs**
```C#
namespace Issue3063 {
    using Sc.Server.Weaver.Schema;
    using Starcounter;
    using System;
    using System.CodeDom.Compiler;
    using System.IO;
    using System.Reflection;
    
    class Program {

        static void Main() {
            var schema = GetWeaverSchemaForApplication(Application.Current);
            schema.DebugOutput(new IndentedTextWriter(Console.Out));
        }

        static DatabaseSchema GetWeaverSchemaForApplication(Application app) {
            // Close your eyes
            var getHostedFile = typeof(Application).GetProperty("HostedFilePath", BindingFlags.NonPublic | BindingFlags.Instance);
            var path = getHostedFile.GetMethod.Invoke(app, null) as string;

            return DatabaseSchema.DeserializeFrom(new DirectoryInfo(Path.GetDirectoryName(path)));
        }
    }
}
```

**Comment by dan31** [comment]
I was thinking of the DiaApp the other day. But it can only work to output classes, not to draw them :) It's just that you visualise graph from metadata via http://mdaines.github.io/viz.js/ or any similar. If one have some time, building such an app is several hours of work.

**Comment by miyconst** [comment]
We can us (d3.js)[https://d3js.org/] for this purpose.
